### PR TITLE
feat(cfl): migrate core commands to artifact projection

### LINKED ISSUE #200: 6/8 feat(cfl): migrate core command outputs to the default artifact contract
## What

Migrate the core `cfl` command surface to the output artifact contract defined in #197, using the shared projection support introduced in #198.

The resulting default artifacts should be agent-oriented and preserve the information an LLM commonly needs to act on Confluence content, including page identity, navigational context, and page content in an agent-usable form.

Ancestor context should be treated as first-class output where it materially affects workflows such as creating child pages, understanding page placement, and navigating a space.

## Why

`cfl` output is often consumed by agents that need to understand page structure and make follow-up decisions, not inspect raw Confluence payloads.

Today, the default outputs expose too much transport and storage detail while under-emphasizing the contextual information that agents actually need. Migrating `cfl` to the default artifact contract will make the tool more useful in its primary usage mode.

## Notes

- This issue is about the `cfl` command surface and default artifacts, not the shared architecture work.
- Commands whose outputs depend on transformed content should rely on the fidelity work in #196 when needed.
- Richer inspection-oriented and source-faithful representations should remain available when explicitly requested under the artifact model defined in #197.

### PR DESCRIPTION
**Pull request by rianjs**
## Summary

- Migrate `page view`, `page list`, `space view`, `space list`, and `search` to artifact projection
- Create `tools/cfl/internal/artifact/` package with PageArtifact, PageListArtifact, SpaceArtifact, SearchResultArtifact
- JSON output now returns intentional artifacts instead of raw API payloads
- Content is XHTML by default until #196 validates markdown conversion

### Artifacts

| Artifact | Fields (agent) | Additional (full) |
|----------|----------------|-------------------|
| PageArtifact | id, title, spaceId, spaceKey, parentId, content | version, createdAt, authorId |
| PageListArtifact | id, title, spaceId, status | version, parentId |
| SpaceArtifact | id, key, name | type, status, description |
| SearchResultArtifact | id, title, type, spaceName, excerpt | url, modifiedAt |

### Deferred to follow-up issues

- `page create`, `page edit`, `page copy` (return minimal responses)
- Ancestors support (requires API enhancement)
- Attachment artifacts

## Test plan

- [x] `make build` passes
- [x] `make test` passes (940 tests in 13 cfl packages)
- [x] `make lint` passes (pre-existing gosec warnings only)
- [ ] Manual: `cfl page view <id> -o json` returns PageArtifact
- [ ] Manual: `cfl page list <space> -o json` returns list format with `_meta`
- [ ] Manual: `cfl space list -o json` returns SpaceArtifact list

Closes #200

### CODE REVIEW

**Comment by rianjs** [comment]
## TDD Assessment

**Verdict: Solid unit coverage on the new artifact layer; command-level JSON wiring is untested.**

---

### What is well-covered

The three new artifact files each have thorough unit tests:

- Agent vs Full mode distinctions are exercised for every artifact type
- Nil pointer guards are tested: nil Version in PageArtifact, nil Description and nil Description.Plain in SpaceArtifact
- Excerpt truncation tests cover ASCII, multi-byte UTF-8, and the empty-excerpt case
- Batch functions (ProjectPageListItems, ProjectSearchResults, ProjectSpaces) have slice length and value spot-checks
- The deleted TestEnrichPageWithSpaceKey is correctly superseded by TestProjectPage

---

### Gaps

**1. Command-level JSON output paths have no tests (most material gap)**

Five files each got a new `if opts.Output == "json"` branch that calls `RenderArtifactList` or `RenderArtifact`. None of these branches are exercised in the existing test suites. The projection functions are correct in isolation, but wiring bugs (wrong mode passed, wrong content extracted, wrong `hasMore` flag) would not be caught.

Affected:
- `tools/cfl/internal/cmd/page/list.go`
- `tools/cfl/internal/cmd/page/view.go`
- `tools/cfl/internal/cmd/search/search.go`
- `tools/cfl/internal/cmd/space/list.go`
- `tools/cfl/internal/cmd/space/view.go`

**2. Content extraction branching in page/view.go is untested**

The new runView code picks between hasStorageContent and hasADFContent to build the content string passed to ProjectPage. The test file has no case covering the JSON output path at all (all JSON-specific test coverage for view was removed with the deleted TestEnrichPageWithSpaceKey and not replaced at the command level).

**3. Nil slice inputs to batch functions**

The commands pass `nil` to the batch functions when results are empty (e.g., `ProjectPageListItems(nil, ...)`). This works because `make([]*T, len(nil))` returns an empty slice, but there is no explicit test for this path in the batch function tests. Low risk, but worth a one-liner.

**4. No regression guard on the JSON output shape**

This replaces raw API payloads with curated artifacts -- a breaking change to the JSON contract. There is no test that serializes to JSON and asserts the final field names or the omitempty behavior (e.g., confirming that a zero Version is omitted vs present). If a downstream consumer depends on the old shape, nothing catches a regression.

**5. Boundary case: excerpt exactly at 200 runes**

The truncation tests cover length > 200 and length < 200 but not length == 200. The boundary produces a rune slice of exactly 200 which should not be truncated, but it is not exercised.

---

### Summary table

| Area | Coverage |
|------|----------|
| Artifact projection logic (agent/full modes) | Good |
| Nil pointer guards in projectors | Good |
| Multi-byte string handling | Good |
| Batch functions | Adequate |
| Command-level JSON wiring | None |
| Content extraction branch (view.go) | None |
| Serialized JSON field shape | None |
| Excerpt boundary at exactly 200 runes | Missing |

---

None of these gaps block merge if the team treats the artifact unit tests as sufficient and considers the JSON output shape not yet stable. The command-level wiring gap (point 1) is the one most likely to harbor a silent bug.

File: tools/cfl/internal/artifact/page.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		SpaceKey: spaceKey,
+		ParentID: p.ParentID,
+		Content:  content,
+	}
+
+	if mode == artifact.Full {
+		if p.Version != nil {
+			art.Version = p.Version.Number
+			if !p.Version.CreatedAt.IsZero() {
+				art.CreatedAt = p.Version.CreatedAt.Format("2006-01-02T15:04:05Z")
```
Comment (monit-reviewer) [inline tools/cfl/internal/artifact/page.go]: 🔵 **Low** (harness-engineering:architecture-reviewer): Hardcoded 'Z' literal in time format string: `p.Version.CreatedAt.Format("2006-01-02T15:04:05Z")`. In Go's time package, 'Z' is emitted verbatim regardless of the time's actual timezone — it is not a timezone layout element. If the Confluence API ever returns a non-UTC timestamp, the output will silently claim UTC while displaying the wrong wall time. Fix: use `p.Version.CreatedAt.UTC().Format(time.RFC3339)` or the layout `"2006-01-02T15:04:05Z07:00"`.

File: tools/cfl/internal/cmd/page/view_test.go [hunk 2]
```diff
@@ -426,21 +426,6 @@ func TestRunView_SpaceLookupFails_Graceful(t *testing.T) {
 	testutil.RequireNoError(t, err)
```
Comment (monit-reviewer) [inline tools/cfl/internal/cmd/page/view_test.go:426]: 🟡 **Medium** (harness-engineering:architecture-reviewer): TestEnrichPageWithSpaceKey (the only test covering the JSON output path in runView) was deleted, but no replacement test exercises the new artifact JSON path through runView. The behavior change — `v.JSON(enrichedPage)` replaced by `v.RenderArtifact(PageArtifact)` — is not covered at the command level. Edge cases like the ADF branch, the empty-content fallback, and correct spaceKey wiring are also untested. The manual test plan items for this path remain unchecked.

File: tools/cfl/internal/cmd/page/view.go [hunk 3]
```diff
@@ -128,9 +128,16 @@ func runView(ctx context.Context, pageID string, opts *viewOptions) error {
 	}
 
 	if opts.Output == "json" {
-		// Enrich JSON output with spaceKey
-		enriched := enrichPageWithSpaceKey(page, spaceKey)
-		return v.JSON(enriched)
+		// Extract content (XHTML storage format by default, until #196 validates markdown)
```
Comment (monit-reviewer) [inline tools/cfl/internal/cmd/page/view.go:131]: 🔵 **Low** (harness-engineering:enforcement-reviewer): When a page body is in neither Storage nor ADF format (e.g., Confluence editor2/Fabric pages), `content` silently defaults to `""` with no log, warning, or distinguishing field. Consumers cannot tell whether the page has no content or whether its content is in an unsupported format. Consider a `contentFormat` field or a stderr debug message so downstream consumers can differentiate these cases.

File: tools/cfl/internal/artifact/search.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+// maxExcerptRunes is the maximum number of runes for excerpt in agent mode.
+const maxExcerptRunes = 200
+
+// ProjectSearchResult projects an api.SearchResult to a SearchResultArtifact.
+func ProjectSearchResult(r *api.SearchResult, mode artifact.Type) *SearchResultArtifact {
+	art := &SearchResultArtifact{
+		ID:        r.Content.ID,
+		Title:     r.Title,
```
Comment (monit-reviewer) [inline tools/cfl/internal/artifact/search.go:27]: 🔵 **Low** (database:sql-reviewer): Excerpt length is checked by creating a rune slice, then a second rune slice is created for the actual truncation. Convert to runes once: `runes := []rune(excerpt); if len(runes) > maxExcerptRunes { ... }` to avoid the redundant allocation.

File: tools/cfl/internal/artifact/page.go [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+)
+
+// PageArtifact represents a content-bearing page artifact for page view.
+type PageArtifact struct {
+	ID        string `json:"id"`
+	Title     string `json:"title"`
+	SpaceID   string `json:"spaceId"`
+	SpaceKey  string `json:"spaceKey,omitempty"`
+	ParentID  string `json:"parentId,omitempty"`
+	Content   string `json:"content"`
```
Comment (monit-reviewer) [inline tools/cfl/internal/artifact/page.go:18]: 🔵 **Low** (harness-engineering:architecture-reviewer): `Version int` uses `json:"version,omitempty"`, so a version value of 0 is silently omitted from full-mode JSON output. Full mode is intended to return the complete record; a missing field and a zero value are indistinguishable to consumers. If version 0 is ever valid (or the API returns it for newly created pages), the field will silently disappear.

File: tools/cfl/internal/artifact/page.go [hunk 6]
```diff
@@ -0,0 +1,86 @@
+// Package artifact provides artifact types for projecting API responses to
+// structured output for CLI commands.
+package artifact
+
+import (
+	"github.com/open-cli-collective/atlassian-go/artifact"
+
+	"github.com/open-cli-collective/confluence-cli/api"
+)
+
```
Comment (monit-reviewer) [inline tools/cfl/internal/artifact/page.go:10]: 🔵 **Low** (harness-engineering:architecture-reviewer): `PageArtifact.Content` carries raw XHTML (or ADF) with no field indicating which format is present. The PR documents this as 'XHTML until #196', but the artifact schema itself gives no signal to consumers. A `contentFormat` field (e.g., `"storage"` / `"adf"` / `"markdown"`) alongside `Content` would allow consumers to handle the format correctly without out-of-band knowledge.

**Review by monit-reviewer — CHANGES_REQUESTED** [review]
## Automated PR Review



**Reviewed commit:** `c7ad6a8`

### Summary

| Reviewer | Findings |
|----------|----------|
| database:sql-reviewer | 1 |
| harness-engineering:architecture-reviewer | 4 |
| harness-engineering:enforcement-reviewer | 1 |

<details>
<summary><strong>database:sql-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `tools/cfl/internal/artifact/search.go:27`
> Excerpt length is checked by creating a rune slice, then a second rune slice is created for the actual truncation. Convert to runes once: `runes := []rune(excerpt); if len(runes) > maxExcerptRunes { ... }` to avoid the redundant allocation.

</details>

<details>
<summary><strong>harness-engineering:architecture-reviewer (4 findings)</strong></summary>

**💡 Suggestion** - `tools/cfl/internal/artifact/page.go:41`
> Hardcoded 'Z' literal in time format string: `p.Version.CreatedAt.Format("2006-01-02T15:04:05Z")`. In Go's time package, 'Z' is emitted verbatim regardless of the time's actual timezone — it is not a timezone layout element. If the Confluence API ever returns a non-UTC timestamp, the output will silently claim UTC while displaying the wrong wall time. Fix: use `p.Version.CreatedAt.UTC().Format(time.RFC3339)` or the layout `"2006-01-02T15:04:05Z07:00"`.

**⚠️ Should Fix** - `tools/cfl/internal/cmd/page/view_test.go:426`
> TestEnrichPageWithSpaceKey (the only test covering the JSON output path in runView) was deleted, but no replacement test exercises the new artifact JSON path through runView. The behavior change — `v.JSON(enrichedPage)` replaced by `v.RenderArtifact(PageArtifact)` — is not covered at the command level. Edge cases like the ADF branch, the empty-content fallback, and correct spaceKey wiring are also untested. The manual test plan items for this path remain unchecked.

**💡 Suggestion** - `tools/cfl/internal/artifact/page.go:18`
> `Version int` uses `json:"version,omitempty"`, so a version value of 0 is silently omitted from full-mode JSON output. Full mode is intended to return the complete record; a missing field and a zero value are indistinguishable to consumers. If version 0 is ever valid (or the API returns it for newly created pages), the field will silently disappear.

**💡 Suggestion** - `tools/cfl/internal/artifact/page.go:10`
> `PageArtifact.Content` carries raw XHTML (or ADF) with no field indicating which format is present. The PR documents this as 'XHTML until #196', but the artifact schema itself gives no signal to consumers. A `contentFormat` field (e.g., `"storage"` / `"adf"` / `"markdown"`) alongside `Content` would allow consumers to handle the format correctly without out-of-band knowledge.

</details>

<details>
<summary><strong>harness-engineering:enforcement-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `tools/cfl/internal/cmd/page/view.go:131`
> When a page body is in neither Storage nor ADF format (e.g., Confluence editor2/Fabric pages), `content` silently defaults to `""` with no log, warning, or distinguishing field. Consumers cannot tell whether the page has no content or whether its content is in an unsupported format. Consider a `contentFormat` field or a stderr debug message so downstream consumers can differentiate these cases.

</details>

*3 info-level observations excluded. Run with `--verbose` to include.*

---
<details>
<summary>Completed in 2m 51s | $1.08</summary>

| Field | Value |
|---|---|
| Reviewers | hybrid-synthesis, database:sql-reviewer, security:code-auditor, harness-engineering:knowledge-reviewer, harness-engineering:enforcement-reviewer, harness-engineering:architecture-reviewer, harness-engineering:legibility-reviewer |
| Engine | claude |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 2m 51s wall · 7m 22s compute (Reviewers: 1m 40s · Synthesis: 1m 07s) |
| Cost | $1.08 |
| Tokens | 449.0k in / 31.4k out |
| Turns | 21 |

</details>

**Comment by rianjs** [comment]
## Response to TDD Assessment

Thanks for the thorough review. Addressed two immediate issues:

1. **`--full` flag collision fixed** - Renamed `page view --full` to `--no-truncate` to avoid shadowing the global artifact `--full` flag.

2. **`limit=0` JSON contract fixed** - `page list --limit 0 -o json` now returns artifact list wrapper with `_meta` instead of bare `[]`.

### Acknowledged gaps (not blocking for this PR):

- **Command-level JSON path tests** - The artifact projection is tested at the artifact package level. Command-level integration tests would add coverage but aren't strictly required for this migration.
- **Nil slice edge case** - Works correctly per Go semantics; explicit test is nice-to-have.
- **JSON output shape regression tests** - The contract change is intentional; old shape is replaced by design.
- **Excerpt boundary at exactly 200 runes** - Edge case, existing tests cover above/below threshold behavior.

File: tools/cfl/internal/artifact/page.go (hunk 1 above)
Comment (rianjs) [reply tools/cfl/internal/artifact/page.go]: Fixed - using .UTC().Format() with proper timezone layout.

File: tools/cfl/internal/cmd/page/view_test.go (hunk 2 above)
Comment (rianjs) [reply tools/cfl/internal/cmd/page/view_test.go:426]: Acknowledged - command-level JSON tests are a nice-to-have; the artifact package tests cover the projection logic. The manual test plan items will be verified before merge.

File: tools/cfl/internal/cmd/page/view.go (hunk 3 above)
Comment (rianjs) [reply tools/cfl/internal/cmd/page/view.go:131]: Deferred to #196 - contentFormat field will be added when markdown becomes the default. For now, the contract is documented as XHTML until then.

File: tools/cfl/internal/artifact/search.go (hunk 4 above)
Comment (rianjs) [reply tools/cfl/internal/artifact/search.go:27]: Fixed - now converts to runes once before checking length.

File: tools/cfl/internal/artifact/page.go (hunk 5 above)
Comment (rianjs) [reply tools/cfl/internal/artifact/page.go:18]: Acknowledged - Confluence versions start at 1, so version 0 is not a valid API response. Keeping omitempty to avoid noise in agent mode. Can revisit if this assumption breaks.

File: tools/cfl/internal/artifact/page.go (hunk 6 above)
Comment (rianjs) [reply tools/cfl/internal/artifact/page.go:10]: Deferred to #196 - will add contentFormat field when we introduce markdown as default. Current contract is XHTML-only (documented in code comments and plan).

File: tools/cfl/internal/cmd/page/view.go [hunk 7]
```diff
@@ -69,9 +69,9 @@ JSON output (--output json) always includes the full body.`,
 
 	cmd.Flags().BoolVar(&opts.raw, "raw", false, "Show raw Confluence storage format (XHTML) instead of markdown")
 	cmd.Flags().BoolVarP(&opts.web, "web", "w", false, "Open in browser instead of displaying")
-	cmd.Flags().BoolVar(&opts.full, "full", false, "Show full content without truncation")
+	cmd.Flags().BoolVar(&opts.noTruncate, "no-truncate", false, "Show full content without truncation")
```
Comment (monit-reviewer) [inline tools/cfl/internal/cmd/page/view.go:72]: 🔵 **Low** (harness-engineering:architecture-reviewer): The `--full` flag was renamed to `--no-truncate` to resolve a collision with the global artifact `--full` flag. The rename is intentional, but existing users or automation passing `cfl page view <id> --full` will break with no deprecation alias or migration notice. This should be called out explicitly in the release notes or changelog for the fix commit.

File: tools/cfl/internal/artifact/search.go [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+import (
+	"github.com/open-cli-collective/atlassian-go/artifact"
+
+	"github.com/open-cli-collective/confluence-cli/api"
+)
+
+// SearchResultArtifact represents a search result artifact.
+type SearchResultArtifact struct {
+	ID         string `json:"id"`
```
Comment (monit-reviewer) [inline tools/cfl/internal/artifact/search.go:11]: 🔵 **Low** (harness-engineering:knowledge-reviewer): `SearchResultArtifact` exposes `SpaceName` (display name) but omits `SpaceKey` (stable lookup identifier). Consumers performing follow-up operations — e.g., listing pages in a space returned by search — need the key, not the display name, and will require a second round-trip to resolve it.

**Review by monit-reviewer — APPROVED** [review]
## Automated PR Review



**Reviewed commit:** `fb20a05` | **Previous:** `c7ad6a8` (incremental)

> **Approved** with 2 non-blocking suggestions below. Address at your discretion.

### Summary

| Reviewer | Findings |
|----------|----------|
| harness-engineering:architecture-reviewer | 1 |
| harness-engineering:knowledge-reviewer | 1 |

<details>
<summary><strong>harness-engineering:architecture-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `tools/cfl/internal/cmd/page/view.go:72`
> The `--full` flag was renamed to `--no-truncate` to resolve a collision with the global artifact `--full` flag. The rename is intentional, but existing users or automation passing `cfl page view <id> --full` will break with no deprecation alias or migration notice. This should be called out explicitly in the release notes or changelog for the fix commit.

</details>

<details>
<summary><strong>harness-engineering:knowledge-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `tools/cfl/internal/artifact/search.go:11`
> `SearchResultArtifact` exposes `SpaceName` (display name) but omits `SpaceKey` (stable lookup identifier). Consumers performing follow-up operations — e.g., listing pages in a space returned by search — need the key, not the display name, and will require a second round-trip to resolve it.

</details>

*4 info-level observations excluded. Run with `--verbose` to include.*

*8 PR discussion threads considered.*

---
<details>
<summary>Completed in 3m 21s | $1.18</summary>

| Field | Value |
|---|---|
| Mode | Re-review · Cycle 2 · Session resumed |
| Reviewers | hybrid-synthesis, database:sql-reviewer, security:code-auditor, harness-engineering:knowledge-reviewer, harness-engineering:enforcement-reviewer, harness-engineering:architecture-reviewer, harness-engineering:legibility-reviewer |
| Engine | claude |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 3m 21s wall · 8m 07s compute (Reviewers: 1m 42s · Synthesis: 1m 20s) |
| Cost | $1.18 |
| Tokens | 317.8k in / 34.6k out |
| Turns | 13 |

</details>
